src/benchmark/tpcds_benchmark.cpp

```diff
@@ -0,0 +1,172 @@
+const std::unordered_set<std::string> filename_blacklist();
+
+}  // namespace
+
+int main(int argc, char* argv[]) {
+  const std::string binary_path = argv[0];
+  const std::string binary_directory = binary_path.substr(0, binary_path.find_last_of("/"));
+
+  auto cli_options = opossum::BenchmarkRunner::get_basic_cli_options("TPC-DS Benchmark");
+
+  // clang-format off
+  cli_options.add_options()
+    ("s,scale", "Database scale factor (1 ~ 1GB)", cxxopts::value<int32_t>()->default_value("1"));
+  // clang-format on
+
+  std::shared_ptr<opossum::BenchmarkConfig> config;
+  int32_t scale_factor{};
+
+  if (opossum::CLIConfigParser::cli_has_json_config(argc, argv)) {
+    // JSON config file was passed in
+    const auto json_config = opossum::CLIConfigParser::parse_json_config_file(argv[1]);
+    scale_factor = json_config.value("scale", 1);
+    config = std::make_shared<opossum::BenchmarkConfig>(
+        opossum::CLIConfigParser::parse_basic_options_json_config(json_config));
+
+  } else {
+    // Parse regular command line args
+    const auto cli_parse_result = cli_options.parse(argc, argv);
+
+    if (CLIConfigParser::print_help_if_requested(cli_options, cli_parse_result)) {
+      return 0;
+    }
+    scale_factor = cli_parse_result["scale"].as<int32_t>();
+
+    config =
+        std::make_shared<opossum::BenchmarkConfig>(opossum::CLIConfigParser::parse_basic_cli_options(cli_parse_result));
+  }
+
+  const std::vector<int32_t> valid_scale_factors{1, 1000, 3000, 10000, 30000, 100000};
+
+  const auto& find_result = std::find(valid_scale_factors.begin(), valid_scale_factors.end(), scale_factor);
+  Assert(find_result != valid_scale_factors.end(),
+         "TPC-DS benchmark only supports scale factor 1 (qualification only), 1000, 3000, 10000, 30000 and 100000.");
+
+  auto context = opossum::BenchmarkRunner::create_context(*config);
+
+  std::cout << "- TPC-DS scale factor is " << scale_factor << std::endl;
+
+  std::string query_path = "third_party/tpcds-result-reproduction/query_qualification";
+  std::string table_path = "resources/benchmark/tpcds/tables";
+
+  Assert(std::filesystem::is_directory(query_path), "Query path (" + query_path + ") has to be a directory.");
+  Assert(std::filesystem::is_directory(table_path), "Table path (" + table_path + ") has to be a directory.");
+  Assert(std::filesystem::exists(std::filesystem::path{query_path + "/01.sql"}), "Queries have to be available.");
+  Assert(std::filesystem::exists(std::filesystem::path{table_path + "/call_center.csv.json"}),
+         "Table schemes have to be available.");
+
+  if (!data_files_available(table_path)) {
+    if (std::filesystem::exists(std::filesystem::path{binary_directory + "/dsdgen"})) {
+      const auto files_setup_return =
+          system(("cd " + binary_directory + " && ./dsdgen -scale " + std::to_string(scale_factor) +
+                  " -dir ../resources/benchmark/tpcds/tables -terminate n -verbose -suffix .csv -f")
```

Comment:
So the hyriseBenchmarkTPCDS can only be run from one directory, right? Which is that? Please add an Assert with an informative message that makes sure that it is run from the right directory. Keep in mind that we have a "resources" symlink in every build dir.